spec/fixtures/api/singleton-data/main.js

```diff
@@ -56,11 +59,21 @@ app.on('second-instance', (event, args, workingDirectory, data, ackCallback) =>
   }
   setImmediate(() => {
     console.log([JSON.stringify(args), JSON.stringify(data)].join('||'));
-    sendAck ? ackCallback(ackObj) : ackCallback();
-    setImmediate(() => {
-      app.exit(0);
-    });
-  });
+    if (!doSetTimeout) {
+      sendAck ? ackCallback(ackObj) : ackCallback();
+      setImmediate(() => {
+        app.exit(0);
+      });
+    } else {
+      setTimeout(() => {
+        sendAck ? ackCallback(ackObj) : ackCallback();
+        // cannot be setImmediate
+        setTimeout(() => {
```

Comment:
When we use `setImmediate` here, the test case fails because the app's exit code is -1 (displayed as a large unsigned integer). I'm not sure what is causing that, yet.